### PR TITLE
RAINCATCH-590 Protractor Tests For fh-wfm-workorder UI module

### PR DESCRIPTION
**Pull request by vchepeli**
To run use following command
```bash
HTTP_HOST=testing-vhxefzk2qmoguwjwsfbpffg7-dev.rmadevxp1.skunkhenry.com/\?url\=http://testing-vhxefzjev54dgaznaqdm6gqp-dev.rmadevxp1.skunkhenry.com HTTP_PORT=80 grunt ui:local
```

### CODE REVIEW

File: tests/login-spec.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  });
+  it('login with correct password', function() {
+    loginPage.commands.selfCheck();
+    loginPage.commands.login('trever', '123');
+  });
+  it('scheduler page is displayed', function() {
+    utils.waitNotPresent(loginPage.selectors.logoutButton);
+    schedulerPage.commands.selfCheck();
+  });
+  xit('open user settings', function() {
```
Comment (TommyJ1994) [inline tests/login-spec.js]: `xit` -> `it`? Few more of these below.

Comment (vchepeli) [inline tests/login-spec.js]: this tests should be skipped atm but implemented in future

Comment (TommyJ1994) [inline tests/login-spec.js]: Ah, i see. :+1:

File: pages/login.po.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+
+var expect = chai.expect;
+
+var LoginPage = function() {
+
+  var selectors = {
+    username: 'form[name="loginForm"] input#username',
```
Comment (twnolan) [inline pages/login.po.js]: Looks good so far but I have just a few comments to garner some discussion about a possible contribution recommendations or style guide.

The protractor style guide suggests avoiding the use of css paths and xpath due to the fact they can make tests brittle due to possible changes over time and provides locator strategies to target specific elements such as by.model and by.binding, which are preferred, and also by.id and by.css (http://www.protractortest.org/#/style-guide). Should would be using these locators or is there an advantage of the use of css paths?

For example here: 

`username: 'form[name="loginForm"] input#username' `

could become:

`username: element(by.model('ctrl.username'))` or `username: element(by.id('username'))`

Also instead of using `$` in the functions, it could also be used here instead:

`username: $('form[name="loginForm"] input#username')`

Comment (vchepeli) [inline pages/login.po.js]: @twnolan we can use binding or models. In my case I use css and it same as by.id or.class to be faster. But we can always change this when binding or model elements are available

File: pages/login.po.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    logoutSideButton: 'md-sidenav button[aria-label$="Logout"]',
+    logoutButton: 'form[name="loginForm"] button[aria-label="Log in"]'
+  };
+
+  var commands = {
+    navigate: function() {
+      return browser.get('#/login');
+    },
+    selfCheck: function() {
+      expect(browser.getLocationAbsUrl()).eventually.to.equal('/login');
```
Comment (twnolan) [inline pages/login.po.js]: Regarding test assertions inside the page object, much of the reading I have done around page objects have suggested to keep assertions in the test files and to keep the page objects as simple and separate objects to provide the data for the tests to use (https://martinfowler.com/bliki/PageObject.html). 

Would we be better adopting this approach and creating a set of utility libraries for each page that uses the page object like our tests do and would provide the means for more complex commands and checking, which should allow for the test files here to be unchanged and avoid code duplication in other test files, while also keeping the boundaries between the page objects and test files?

File: tests/login-spec.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+var loginPage = require('../pages/login.po');
+var schedulerPage = require('../pages/scheduler.po');
+
+var utils = require('../utils/utils');
+
+describe('Login into Portal App', function() {
+
+  before('navigate to login page', function() {
+    loginPage.commands.navigate();
+    var progress = 'md-progress-circular';
```
Comment (twnolan) [inline tests/login-spec.js]: As this is a page element, could/should it be implemented inside the page object itself?

File: utils/utils.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+
+//var expect = chai.expect;
+
+module.exports.waitNotPresent = function(selector) {
```
Comment (twnolan) [inline utils/utils.js]: I really like these utility methods as they provide functionality that protractor does not possess but needs, but would it be beneficial to allow a timeout value to be passed as an argument and defaults to 10000 if no value was passed? 

It might help if some elements were slower than others to load or disappear, or performance was reduced by load and we needed to adjust times for tests.

File: pages/login.po.js (hunk 2 above)
Comment (vchepeli) [inline pages/login.po.js]: One its merged feel free to send PR with model/binding elements

File: pages/login.po.js (hunk 3 above)
Comment (vchepeli) [inline pages/login.po.js]: this one is assertion commands and has nothing related to page object navigation commands. And this command should check if element is there otherwise throw assertion error

File: tests/login-spec.js (hunk 4 above)
Comment (vchepeli) [inline tests/login-spec.js]: Yep this one should be implemented but its common one element. Now I leave it like its and will see if need to add it to page object

File: utils/utils.js (hunk 5 above)
Comment (vchepeli) [inline utils/utils.js]: In this case dont care about use dynamic timouts. **It should fail fast**. Otherwise we have performance issues or wrong configuration. And by model user can wait maximum 2secs for element to appear. Here its 5x times more

File: pages/login.po.js (hunk 2 above)
Comment (TommyJ1994) [inline pages/login.po.js]: +1 on selecting elements in the UI based on the model name.
